torchao/quantization/quant_api.py

```diff
@@ -222,6 +224,9 @@ def _replace_with_custom_fn_if_matches_filter(
     Returns:
         None
     """
+    # If model is Float8Linear, convert it to Linear before moving forward
+    if isinstance(model, Float8Linear):
+        model = dequantize_float8_training(model)
```

Comment:
can you just move your code snippet from the other file here:

```
if isinstance(model, Float8Linear):
    with torch.device("meta"):
        new_module = nn.Linear(model.in_features, model.out_features)
    new_module.weight = model.weight
    new_module.bias = model.bias        
    model = new_module
```

and not need any changes to `torchao/float8`?

Reply:
@vkuzo what do you think about having dequantizing a model as a separate API? it feels a bit weird to have this logic in `_replace_with_custom_fn_if_matches_filter` which is supposed to be a simple module replacement function I feel.

Reply:
my 5c - 
1. we can make it work now without adding **any** public APIs, with minimal increase in complexity
2. if it's important to have a public API for "remove low precision training from a model", we can have that conversation in parallel

wdyt

Reply:
As per the offline discussion with @jerryzh168, we should have a separate exposed API to convert Float8Linear to Linear. The flow would then be:

-> Float8 training
-> Dequantize Float8Linear to Linear
-> quantize_()

Reply:
the motivation for adding a new API is making the dequantizing step more explicit for user, instead of hide it in a module replacement function.

but agree this can happen in parallel. also it's probably not worth spending time to discuss as of now, and wait until there are more use cases might be better